### PR TITLE
Set each partition weights in waged configs

### PR DESCRIPTION
**Pull request by Arun-LinkedIn**

### CODE REVIEW

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java [hunk 1]
```diff
@@ -1106,6 +1118,64 @@ private void setInstanceConfig(String dcName, String instanceName, String resour
...
+          String partitionName = replicaConfigEntry.getKey();
+          long capacityInGB = replicaConfigEntry.getValue().getReplicaCapacityInBytes() / 1024 / 1024 / 1024;
+          partitionCapacityMap.put(partitionName, new HashMap<>());
+          partitionCapacityMap.get(partitionName)
+              .put(DISK_KEY, Math.toIntExact(capacityInGB) + PARTITION_BUFFER_CAPACITY_FOR_INDEX_FILES_IN_GB);
+        }
+      }
+    }
+    partitionCapacityMap.put(DEFAULT_PARTITION_KEY, new HashMap<>());
+    partitionCapacityMap.get(DEFAULT_PARTITION_KEY).put(DISK_KEY, clusterConfigFields.partitionDiskWeightInGB);
```
Comment (justinlin-linkedin) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:1170]: should add PARTITION_BUFFER_CAPACITY_FOR_INDEX_FILES_IN_GB.

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java [hunk 2]
```diff
@@ -1106,6 +1118,64 @@ private void setInstanceConfig(String dcName, String instanceName, String resour
...
+      Map<String, DataNodeConfig.DiskConfig> diskConfigs = dataNodeConfig.getDiskConfigs();
+      for (DataNodeConfig.DiskConfig diskConfig : diskConfigs.values()) {
+        Map<String, DataNodeConfig.ReplicaConfig> replicaConfigs = diskConfig.getReplicaConfigs();
+        // Get replica configs for each replica on the disk
+        for (Map.Entry<String, DataNodeConfig.ReplicaConfig> replicaConfigEntry : replicaConfigs.entrySet()) {
+          String partitionName = replicaConfigEntry.getKey();
+          long capacityInGB = replicaConfigEntry.getValue().getReplicaCapacityInBytes() / 1024 / 1024 / 1024;
+          partitionCapacityMap.put(partitionName, new HashMap<>());
+          partitionCapacityMap.get(partitionName)
+              .put(DISK_KEY, Math.toIntExact(capacityInGB) + PARTITION_BUFFER_CAPACITY_FOR_INDEX_FILES_IN_GB);
```
Comment (justinlin-linkedin) [inline ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java:1165]: we don't have to add this partition when its weight is the same as the default one.

Comment (Arun-LinkedIn) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java]: Done.

File: ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java (hunk 1 above)
Comment (Arun-LinkedIn) [reply ambry-clustermap/src/main/java/com/github/ambry/clustermap/HelixBootstrapUpgradeUtil.java]: Thank you for catching! Done

**Review by justinlin-linkedin — APPROVED** [review]

**Pull request merged**